edgelet/hsm-sys/azure-iot-hsm-c/src/certificate_info.c

```diff
@@ -214,20 +214,26 @@ static time_t tm_to_utc(const struct tm *tm)
 time_t get_utc_time_from_asn_string(const unsigned char *time_value, size_t length)
 {
     time_t result;
-    char temp_value[TEMP_DATE_LENGTH];
-    size_t temp_idx = 0;
-    struct tm target_time;
-    uint32_t numeric_val;
 
-    memset(&target_time, 0, sizeof(target_time));
-    memset(temp_value, 0, TEMP_DATE_LENGTH);
-    if (length != TIME_FIELD_LENGTH)
+    if (time_value == NULL)
+    {
+        LogError("Parse time error: Invalid time_value buffer");
+        result = 0;
+    }
+    else if (length != TIME_FIELD_LENGTH)
     {
         LogError("Parse time error: Invalid length field");
         result = 0;
     }
     else
     {
+        char temp_value[TEMP_DATE_LENGTH];
+        size_t temp_idx = 0;
+        struct tm target_time;
+        uint32_t numeric_val;
+
+        memset(&target_time, 0, sizeof(target_time));
+        memset(temp_value, 0, TEMP_DATE_LENGTH);
```

Comment:
This is not test. :) Can you add this to the description?

Reply:
okay done